src/main/kotlin/com/hibob/academy/service/PetService.kt

```diff
@@ -25,4 +25,16 @@ class PetService(private val petDao: PetDao) {
             throw BadRequestException("No pet with that id")
         }
     }
+
+    fun countPetsByType(companyId: Long): Map<String, Int> {
+        return petDao.countPetsByType(companyId)
+    }
+
+    fun adoptMultiple(pets: List<UUID>, ownerId: UUID, companyId: Long): Int {
+        return pets.count { pet -> petDao.assignOwnerIdToPet(pet, ownerId) == 1 }
```

Comment:
There is another way to update multiple at once - which is with better performance
In the dao you can update set owner id where petId in the ids list
You can change it later in another PR... but it would be better